src/components/settings/LocationSettings.js

```diff
@@ -115,13 +115,15 @@ export const LocationSettings = () => {
           </RegularText>
         </Wrapper>
         <Collapsible collapsed={!showMap}>
-          <WebViewMap
-            mapCenterPosition={{ lat: 51.1657, lng: 10.4515 }} // center of germany
+          <Map
             locations={locations}
-            onMessageReceived={(msg) => {
-              if (msg.event === 'onMapClicked') {
-                setSelectedPosition(msg.payload.touchLatLng);
-              }
+            mapCenterPosition={{ lat: 51.1657, lng: 10.4515 }} // center of germany
+            onMapPress={({ nativeEvent }) => {
+              const coordinate = {
+                lat: nativeEvent.coordinate.latitude,
+                lng: nativeEvent.coordinate.longitude
+              };
+              setSelectedPosition(coordinate);
```

Comment:
let's set directly, as we do not reuse the created `coordinate` variable - we do not need it:

```js
setSelectedPosition({
  lat: nativeEvent.coordinate.latitude,
  lng: nativeEvent.coordinate.longitude
});
```

Reply:
https://github.com/ikuseiGmbH/smart-village-app-app/pull/600/commits/01da2404c7d615eaf0c914f270b11636fd22eb69